src/app/tests/TestSceneTable.cpp

```diff
@@ -0,0 +1,852 @@
+                serialisedBytes.reduce_size(99); // Used memory for Color Control TLV
+                break;
+            default:
+                break;
+            }
+        }
+        return err;
+    }
+
+    /// @brief Simulates EFS being applied to a scene, here just validates that the data is as expected, no action taken by the
+    /// "cluster"
+    /// @param endpoint target endpoint
+    /// @param cluster  target cluster
+    /// @param serialisedBytes Data from nvm
+    /// @param timeMs transition time in ms
+    /// @return CHIP_NO_ERROR if value as expected, CHIP_ERROR_INVALID_ARGUMENT otherwise
+    CHIP_ERROR
+    ApplyScene(EndpointId endpoint, ClusterId cluster, ByteSpan & serialisedBytes, TransitionTimeMs timeMs) override
+    {
+        CHIP_ERROR err = CHIP_ERROR_INVALID_ARGUMENT;
+
+        // Takes values from cluster in Endpoint 1
+        if (endpoint == TEST_ENDPOINT1)
+        {
+            switch (cluster)
+            {
+            case ON_OFF_CID:
+                if (!memcmp(serialisedBytes.data(), OO_buffer, serialisedBytes.size()))
+                {
+                    err = CHIP_NO_ERROR;
+                }
+                break;
+            case LV_CTR_CID:
+                if (!memcmp(serialisedBytes.data(), LC_buffer, serialisedBytes.size()))
+                {
+                    err = CHIP_NO_ERROR;
+                }
+                break;
+            default:
+                break;
+            }
+        }
+
+        // Takes values from cluster in Endpoint 2
+        if (endpoint == TEST_ENDPOINT2)
+        {
+            switch (cluster)
+            {
+            case ON_OFF_CID:
+                if (!memcmp(serialisedBytes.data(), OO_buffer, serialisedBytes.size()))
+                {
+                    err = CHIP_NO_ERROR;
+                }
+                break;
+            case CC_CTR_CID:
+                if (!memcmp(serialisedBytes.data(), CC_buffer, serialisedBytes.size()))
+                {
+                    err = CHIP_NO_ERROR;
+                }
+                break;
+            default:
+                break;
+            }
+        }
+
+        return CHIP_NO_ERROR;
+    }
+};
+
+static chip::TestPersistentStorageDelegate testStorage;
+static SceneTableImpl sSceneTable;
+static TestSceneHandler sHandler;
+
+void ResetSceneTable(SceneTable * sceneTable)
+{
+    sceneTable->RemoveFabric(kFabric1);
+    sceneTable->RemoveFabric(kFabric2);
+}
+
+void TestHandlerRegistration(nlTestSuite * aSuite, void * aContext)
+{
+    SceneTable * sceneTable = &sSceneTable;
+    TestSceneHandler tmpHandler[scenes::kMaxSceneHandlers];
+
+    for (uint8_t i = 0; i < scenes::kMaxSceneHandlers; i++)
+    {
+        NL_TEST_ASSERT(aSuite, sceneTable->mNumHandlers == i);
+        NL_TEST_ASSERT(aSuite, CHIP_NO_ERROR == sceneTable->RegisterHandler(&tmpHandler[i]));
+        printf("Handler : %d | Address : %p \n", i, &tmpHandler[i]);
```

Comment:
Do not `printf` directly in tests. Use `ChipLogXxx`

Reply:
Printf were removed in latest commit.